### PR TITLE
[FrameworkBundle] Fix template location for PHP templates

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14804 |
| License | MIT |
| Doc PR | - |
- [x] improve the test to cover logical path & filesystem path
- [x] Add a new test case and fix the path to the template

As the first commit only enchanced the test, and the second commit fixed the bug, it's best to review them seperately.

### CODE REVIEW

**Comment by weaverryan** [comment]
@jakzal I see you've enhanced the test - but I'm confused (in part, because I honestly haven't spent much time looking into the template name parsing stuff): this doesn't contain a fix for #14804 or add a failing test case. Was this just a step 1 to enhance to test? Or am I missing something (or perhaps I haven't answered your questions well yet on #14804).

Thanks for taking this on - I see PR's from you fixing issues. It's awesome.

**Comment by jakzal** [comment]
@weaverryan indeed. That's why it's marked WIP. I updated PR's description.

Few test cases are either wrong ("name.twig" and "name"), or we should treat them differently then templates that follow the "name.html.twig" pattern.

**Comment by jakzal** [comment]
Are "name.twig" and "name" even valid template names?  I went with the second option for now to avoid changing existing cases.

**Comment by weaverryan** [comment]
@jakzal I don't know if `name` or `name.twig` should be valid (`name.twig` should in my opinion). Best to keep these as they are for BC.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Templating/TemplateNameParserTest.php [hunk 1]
```diff
@@ -43,30 +43,33 @@ protected function tearDown()
...
+            array(':Post:index.html.php', ':Post:index.html.php', 'views/Post/index.html.php', new TemplateReference('', 'Post', 'index', 'html', 'php')),
+            array('::index.html.php', '::index.html.php', 'views/index.html.php', new TemplateReference('', '', 'index', 'html', 'php')),
+            array('index.html.php', '::index.html.php', 'views/index.html.php', new TemplateReference('', '', 'index', 'html', 'php')),
+            array('FooBundle:Post:foo.bar.index.html.php', 'FooBundle:Post:foo.bar.index.html.php', '@FooBundle/Resources/views/Post/foo.bar.index.html.php', new TemplateReference('FooBundle', 'Post', 'foo.bar.index', 'html', 'php')),
+            array('/path/to/section/name.php', '/path/to/section/name.php', '/path/to/section/name.php', new BaseTemplateReference('/path/to/section/name.php', 'php')),
+            array('name.twig', 'name.twig', 'name.twig', new BaseTemplateReference('name.twig', 'twig')),
+            array('name', 'name', 'name', new BaseTemplateReference('name')),
+            array('default/index.html.php', '::default/index.html.php', 'views/default/index.html.php', new TemplateReference(null, null, 'default/index', 'html', 'php')),
```
Comment (weaverryan) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Templating/TemplateNameParserTest.php]: So this last entry is the only one that was really added or changed at all, right?

What I don't understand yet is why `default/index.html.twig` seems to have always pointed to `::default/index.html.twig` but `default/index.html.php` did not work correctly (and it only starts working "incorrectly" once both the twig and php engines are enabled).

**Comment by jakzal** [comment]
status: needs review

**Comment by jakzal** [comment]
ping @symfony/deciders

**Comment by xabbuh** [comment]
I have some hard time wrapping my head around these changes, but the result looks like what we expect.

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request merged**